src/services/newsletter.ts

```diff
@@ -40,22 +40,42 @@ export async function newsletterHandler(
 
   sentry.setUser({ email });
 
-  const response = await fetch(MAILERLITE_API, {
-    method: "POST",
-    body: JSON.stringify({ email }),
-    headers: {
-      "Content-Type": "application/json",
-      "X-MailerLite-ApiKey": MAILERLITE_API_KEY,
-    },
-  });
+  let response: Response;
+  let data: Record<string, any>;
 
-  if (!response.ok) {
+  try {
+    response = await fetch(MAILERLITE_API, {
+      method: "POST",
+      body: JSON.stringify({ email }),
+      headers: {
+        "Content-Type": "application/json",
+        "X-MailerLite-ApiKey": MAILERLITE_API_KEY,
+      },
+    });
+  } catch (err: any) {
     throw new ServiceError(
       "Could not subscribe",
       NewsletterErrorType.SUBSCRIPTION
     );
   }
 
+  if (!response.ok) {
+    try {
+      data = await response.json<Record<string, any>>();
+    } catch (err: any) {
+      throw new ServiceError(
+        "Could not parse error",
```

Comment:
We still need to tell user that their action went wrong
```suggestion
        "Could not subscribe (unknown error)",
```